platform/v1alpha1/types.go

```diff
@@ -0,0 +1,76 @@
+package v1alpha1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+var (
+	TypeSourced = "Sourced"
+	TypeApplied = "Applied"
+
+	ReasonSourceFailed     = "SourceFailed"
+	ReasonSourceSuccessful = "SourceSuccessful"
+	ReasonApplyFailed      = "ApplyFailed"
+	ReasonApplySuccessful  = "ApplySuccessful"
+)
+
+// PlatformOperatorSpec defines the desired state of PlatformOperator.
+type PlatformOperatorSpec struct {
+	// PackageName contains the desired OLM-based Operator package name
+	// that is defined in an existing CatalogSource resource in the cluster.
+	//
+	// This configured package will be managed with the cluster's lifecycle.
+	// +kubebuilder:validation:Required
+	// +required
+	PackageName string `json:"packageName"`
+}
+
+// ActiveBundleDeployment references a BundleDeployment resource.
+type ActiveBundleDeployment struct {
+	// Name is the metadata.name of the referenced BundleDeployment object.
+	// +kubebuilder:validation:Required
+	// +required
+	Name string `json:"name"`
+}
+
+// PlatformOperatorStatus defines the observed state of PlatformOperator
+type PlatformOperatorStatus struct {
+	// Conditions represent the latest available observations of a platform operator's current state.
+	// +optional
+	Conditions []metav1.Condition `json:"conditions,omitempty"`
+	// ActiveBundleDeployment is the reference to the BundleDeployment resource that's
+	// being managed by this PO resource.
+	// +optional
+	ActiveBundleDeployment ActiveBundleDeployment `json:"activeBundleDeployment,omitempty"`
+}
+
+// +genclient
+// +genclient:nonNamespaced
+//+kubebuilder:object:root=true
+//+kubebuilder:subresource:status
+//+kubebuilder:resource:scope="Cluster",shortName={"platform","platforms"}
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// PlatformOperator is the Schema for the PlatformOperators API.
+//
+// Compatibility level 4: No compatibility is provided, the API can change at any point for any reason. These capabilities should not be used by applications needing long term support.
+// +openshift:compatibility-gen:level=4
+type PlatformOperator struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	Spec   PlatformOperatorSpec   `json:"spec,omitempty"`
```

Comment:
Do we need a required kubebuilder marker here?

Reply:
Good catch, we just need to remove the `omitempty` text from the json reflection and it'll be marked as required in the CRD.